.github/workflows/build_and_deploy.yml

```
@@ -0,0 +1,131 @@
+
+      - name: Set current env variable
+        run: |
+          if [[ $BRANCH == 'main' ]]; then
+              echo "CURRENT_ENV=production" >> "$GITHUB_ENV"
+          else
+              echo "CURRENT_ENV=staging" >> "$GITHUB_ENV"
+          fi
+
+      - name: Populate outputs
+        id: populate-outputs
+        run: |
+            echo "env_name=${{ env.CURRENT_ENV }}" >> $GITHUB_OUTPUT
+            echo "container_name=alex-ic-${{ env.CURRENT_ENV }}" >> $GITHUB_OUTPUT
+            echo "ecs_cluster=alex-ic-${{ env.CURRENT_ENV }}-ecs-cluster" >> $GITHUB_OUTPUT
+            echo "ecs_service=alex-ic-${{ env.CURRENT_ENV }}-ecs-service" >> $GITHUB_OUTPUT
+            echo "ecs_task_definition=alex-ic-${{ env.CURRENT_ENV }}-service" >> "$GITHUB_OUTPUT"
+
+  build_image:
+    needs: shared_variables
+    name: Build the Docker image
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    outputs:
+      image: ${{ steps.build-image.outputs.image }}
+
+    steps:
+      - name: Debug
+        run: |
+          echo "ECR_REPOSITORY=${{ env.ECR_REPOSITORY }}"
+          echo "REGION=${{ env.REGION }}"
+          echo "env_name=${{ needs.shared_variables.outputs.env_name }}"
+          echo "container_name=${{ needs.shared_variables.outputs.container_name }}"
+          echo "ecs_cluster=${{ needs.shared_variables.outputs.ecs_cluster }}"
+          echo "ecs_service=${{ needs.shared_variables.outputs.ecs_service }}"
+          echo "ecs_task_definition=${{ needs.shared_variables.outputs.ecs_task_definition }}"
+      
+      - name: ENV variables
+        uses: FranzDiebold/github-env-vars-action@v2.7.0
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1.7.0
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.REGION }}
+          mask-aws-account-id: 'no'
+
+      - name: Login to AWS ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1.5.3
+
+      - name: Build, tag, and push image to AWS ECR
+        id: build-image
+        env:
+          SHARED_ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        run: |
+          docker build -t $SHARED_ECR_REGISTRY/$ECR_REPOSITORY:$CI_REF_NAME_SLUG-$CI_SHA_SHORT -t $SHARED_ECR_REGISTRY/$ECR_REPOSITORY:${{ needs.shared_variables.outputs.env_name }} .
+          docker push -a $SHARED_ECR_REGISTRY/$ECR_REPOSITORY
+          echo "image=$SHARED_ECR_REGISTRY/$ECR_REPOSITORY:$CI_REF_NAME_SLUG-$CI_SHA_SHORT" >> $GITHUB_OUTPUT
+  
+  deploy:
+    needs: [build_image, shared_variables]
+    name: Deploy the built image
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1.6.1
```

how about using the same version as in the `build_image` step
```suggestion
        uses: aws-actions/configure-aws-credentials@v1.7.0
```

---

All versions were upgraded here: https://github.com/Nihisil/nimble-devops-ic-app/pull/12/commits/c16ac1f00ce5f796ad404c21d5afadaf45f5e032